src/rpdk/core/project.py

```diff
@@ -293,6 +296,141 @@ def submit(
                     f, endpoint_url, region_name, role_arn, use_role, set_default
                 )
 
+    def generate_docs(self):
+        # generate the docs folder that contains documentation based on the schema
+        docs_path = "{}/docs".format(self.root)
+
+        if not self.type_info or not self.schema or "properties" not in self.schema:
+            LOG.warning(
+                "Could not generate schema docs due to missing type info or schema"
+            )
+            return
```

Comment:
can this actually happen once the schema has been validated?

Reply:
Probably not, but I wanted this for a specific test to fail gracefully. Docs generation shouldn't stop a `cfn generate` in theory if it's passed validation.